tests/integration/test_agent.py

```diff
@@ -18,8 +19,8 @@
 )
 def test_write_simple_script():
     task = "Write a shell script 'hello.sh' that prints 'hello'. Do not ask me for confirmation at any point."
-    final_agent_state = asyncio.run(main(task, exit_on_message=True))
-    assert final_agent_state == AgentState.FINISHED
+    final_state: State = asyncio.run(main(task, exit_on_message=True))
+    assert final_state.agent_state == AgentState.STOPPED
```

Comment:
@li-boxuan Is this OK? I think for now, the agent can end up with two possible final state: `STOPPED` or `ERROR` - `FINISH` will typically lead to `STOPPED`, right?

Reply:
Looks good. Thx!